scam-detector-py/src/agent.py

```diff
@@ -936,6 +938,110 @@ def emit_new_fp_finding(w3) -> list:
+# scammer labels for those addresses. The retrieved data is used to update the dictionary.
+def update_reactive_likely_fps(current_date) -> list:
+    logging.info(f"{BOT_VERSION}: update reactive likely fps called")
+
+    global REACTIVE_LIKELY_FPS
+
+    source_id = '0x47c45816807d2eac30ba88745bf2778b61bc106bc76411b520a5289495c76db8' if Utils.is_beta() else '0x1d646c4045189991fdfd24a66b192a294158b839a6ec121d740474bdacb3ab23'
+    response = None
+    page_size = 2000
+    starting_cursor = None
+    should_retry_from_error = False
+   
+    # Calculate the time 89 days ago (Querying alerts from 90 days ago results in an error)
+    eighty_nine_days_ago = current_date - timedelta(days=89)
+
+    # Calculate the start and end timestamps for the desired hour as milliseconds ago
+    milliseconds_ago = (current_date - eighty_nine_days_ago).total_seconds() * 1000  # Convert to milliseconds
+    start_milliseconds_ago = int(milliseconds_ago)
+    end_milliseconds_ago = int(milliseconds_ago - 3600 * 1000)  # 3600 seconds in an hour, converted to milliseconds
+   
+    alerts = []
+    while True:
+        if response and response.page_info and response.page_info.has_next_page:
+            starting_cursor = {
+                'alertId': response.page_info.end_cursor.alert_id,
+                'blockNumber': response.page_info.end_cursor.block_number
+            }
+            should_retry_from_error = False
+        try:
+            query = {
+                "bot_ids": [source_id],
+                "created_since": start_milliseconds_ago,
+                "created_before": end_milliseconds_ago,
+                "starting_cursor": starting_cursor,
+                "chain_id": CHAIN_ID,
+                "first": page_size
+            }
+            response = get_alerts(query)
+            alerts.extend(response.alerts)
+        except Exception as e:
+            if  ((isinstance(e, AttributeError) and 'NoneType' in str(e)) or (isinstance(e, Exception) and "Internal server error" in str(e))):
+                # Reduce the page size in order to reduce the response size and try again
+                page_size = math.floor(page_size / 2)
+                should_retry_from_error = page_size > 1
+            else:               
+                logging.warning(f"{BOT_VERSION}: update reactive likely fps (get_alerts error): {e} - {traceback.format_exc()}")
+                Utils.ERROR_CACHE.add(Utils.alert_error(str(e), "agent.update_reactive_likely_fps.internal1", traceback.format_exc()))
+        if (not should_retry_from_error and response.page_info.end_cursor.alert_id == ""):
+            break
+        
+    logging.info(f"{BOT_VERSION}: update reactive likely fps (alerts count): {len(alerts)}")
+    
+    unique_scammers_list = Utils.process_past_alerts(alerts, REACTIVE_LIKELY_FPS, BOT_VERSION)
+    labels = []
+    page_size = 1000
+    batch_size = 200
+    for i in range(0, len(unique_scammers_list), batch_size):
```

Comment:
i am not sure why you query alerts first and then labels; could you just query labels directly?

Reply:
@christian-forta, querying all scammer labels directly seemed to be returning too many results and taking much more time compared to this approach. Is there a better filtering for the labels query?

Reply:
i am not sure why that is? are you not applying the same time range? So you would be getting the same amount of data either way, but querying now twice (once for alerts; once for labels)?

Reply:
@christian-forta, I forgot to mention that for the scammers fetched, we're querying (and then removing) _all_ labels, not only those of the specific time range (In get_labels I use only created_since, not created_after).

If I did that without specifying the entities, then I would practically fetch all scammer labels ever generated since the start timestamp, which is an API call that doesn't finish at least in a reasonable time